src/types.ts

```diff
@@ -943,6 +963,41 @@ export class ZodString extends ZodType<string, ZodStringDef, string> {
           });
           status.dirty();
         }
+      } else if (check.kind === "jwt") {
+        const parts = input.data.split(".");
+        if (parts.length !== 3) {
+          ctx = this._getOrReturnCtx(input, ctx);
+          addIssueToContext(ctx, {
```

Comment:
structure the code to avoid multiple calls to `addIssueToContext` for bundle size reasons